virl/files/mitaka.vinstall.py

```diff
@@ -474,51 +474,39 @@ def building_salt_all():
-        with open(("/tmp/foo"), "w") as salt_grain:
-            salt_grain.write("""{""")
-            for key, value in (safeparser.items('DEFAULT')):
-                if key == 'domain': salt_grain.write(""" 'domain_name': '{value}',""".format(key=key,value=value))
-                if value.lower() == 'true' or value.lower() == 'false':
-                    salt_grain.write(""" '{key}': {value} ,""".format(key=key,value=value))
-                else:
-                    salt_grain.write(""" '{key}': '{value}',""".format(key=key,value=value))
-            if cml:
-                salt_grain.write("""  'cinder_enabled': False ,""")
-            else:
-                if cinder_device or cinder_file:
-                    salt_grain.write("""  'cinder_enabled': True ,""")
-                else:
-                    salt_grain.write("""  'cinder_enabled': False ,""")
-            if not uwm_port == '14000':
-                salt_grain.write("""  'uwm_url': 'http://{0}:{1}',""".format(public_ip,uwm_port))
-            salt_grain.write(""" 'neutron_extnet_id': '{neutid}',""".format(neutid=neutron_extnet_id))
-            salt_grain.write(""" 'service_id': '{serviceid}',""".format(serviceid=service_tenid))
-            salt_grain.write(""" 'OS_AUTH_URL': '{auth_url}',""".format(auth_url=keystone_auth_url))
-            if mitaka:
-                salt_grain.write("""  'kilo': False ,""")
-            salt_grain.write(""" 'admin_id': '{adminid}'""".format(adminid=admin_tenid))
-            salt_grain.write("""}""")
-        with open(("/tmp/foo"), "r") as salt_grain_read:
-            subprocess.call(['sudo', 'salt-call', '--local','grains.setvals', salt_grain_read.read() ])
-        subprocess.call(['sudo', 'rm', '-f', '/tmp/foo'])
+        grains_json = json.dumps(grains)
+        subprocess.call(['sudo', 'salt-call', '--local', 'grains.setvals', grains_json])
     else:
-        with open(("/tmp/grains"), "w") as grains:
-            # for section_name in safeparser.sections():
-            if cinder_device or cinder_file:
-                grains.write("""  cinder_enabled: True\n""")
-            else:
-                grains.write("""  cinder_enabled: False\n""")
-            if not uwm_port == '14000':
-                grains.write("""  uwm_url: http://{0}:{1}\n""".format(public_ip,uwm_port))
-            if mitaka:
-                grains.write("""  kilo: False\n""")
-            grains.write("""  OS_AUTH_URL: {1}\n""".format(keystone_auth_url))
-
-            for name, value in safeparser.items('DEFAULT'):
-                grains.write("""  {name}: {value}\n""".format(name=name, value=value))
-            grains.write("""  neutron_extnet_id: {neutid}\n""".format(neutid=neutron_extnet_id))
-        subprocess.call(['sudo', 'mv', '-f', ('/tmp/grains'), '/etc/salt'])
+        grains_yaml = yaml.safe_dump(grains, default_flow_style=False)
+        with open(('/etc/salt/grains'), 'w') as grains_file:
+            grains_file.write(grains_yaml)
+
```

Comment:
Looks complicated, is not move sufficient? Do we have example for such safe update?